File: zema_store_mobile/lib/models/playlist.dart
import 'models.dart';

class Playlist{


  Playlist({
    required this.user_id,
    required this.title,
    required this.songs,
});
  final String user_id;
  final String title;
  final List<Song> songs;

  factory Playlist.fromJson(Map<String,dynamic> json){
    return Playlist(user_id: json['user_id'], title: json['title'], songs: json['songs']);
  }
}

File: zema_store_backend/src/models/mongoose/playlist.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IPlaylistDocument extends Document {
  userId: mongoose.Schema.Types.ObjectId;
  title: String;
  songs: Array<mongoose.Schema.Types.ObjectId>;
  createdAt: Date;
  updatedAt: Date;
}

const playlistSchema: Schema<IPlaylistDocument> = new Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: "CustomerProfile",
    },
    title: {
      type: String,
      required: true,
      default: "custom playlist",
    },
    songs: {
      type: [mongoose.Schema.Types.ObjectId],
      ref: "Song",
      required: true,
    },
  },
  { timestamps: true, toJSON: { virtuals: true }, toObject: { virtuals: true } }
);

const Playlist = mongoose.model<IPlaylistDocument>("Playlist", playlistSchema);

playlistSchema.virtual("id").get(function (this: IPlaylistDocument) {
  return this._id.toHexString();
});

export default Playlist;
